File: homework_ui_python_try/Pages/BasePage.py
from selenium.webdriver import ActionChains
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait


class BasePage:

    def __init__(self, driver):
        self._driver = driver
        self._wait = WebDriverWait(self._driver, 10)

    def click(self, webelement):
        el = self._wait.until(expected_conditions.element_to_be_clickable(webelement))
        el.click()

    def get_text(self, webelement):
        el = self._wait.until(expected_conditions.visibility_of_element_located(webelement))
        return el.text

    def fill_text(self, webelement, txt):
        el = self._wait.until(expected_conditions.element_to_be_clickable(webelement))
        el.clear()
        el.send_keys(txt)

    def clear_text(self, webelement):
        el = self._wait.until(expected_conditions.element_to_be_clickable(webelement))
        el.clear()

    def submit(self, webelement):
        webelement.submit()

    def move_to_element(self, webelement):
        action = ActionChains(self._driver)
        self._wait.until(expected_conditions.visibility_of(webelement))
        action.move_to_element(webelement).perform()

File: homework_ui/src/pageobjects/BasePage.java
package pageobjects;

import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.PageFactory;

public class BasePage {
	WebDriver driver;

	public BasePage(WebDriver driver) {
		this.driver = driver;
		PageFactory.initElements(driver, this);

	}

	public void fillText(WebElement el, String text) {
		clear(el);
		el.sendKeys(text);
		sleep(1000);
	}

	public void click(WebElement el) {
		el.click();
		sleep(1000);
	}

	public void clear(WebElement el) {
		el.clear();
		sleep(1000);
	}

	public void sleep(int mills) {
		try {
			Thread.sleep(mills);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
	}

	public String getText(WebElement el) {
		return el.getText();
	}

	public void submit(WebElement el) {
		el.submit();
	}


	public void alertOK() {
		driver.switchTo().alert().accept();
	}

	public void alertOK(String text) {
		driver.switchTo().alert().sendKeys(text);
		driver.switchTo().alert().accept();
	}

	public void moveToElement(WebElement el) {
		Actions action = new Actions(driver);
		action.moveToElement(el).build().perform();
	}
	
	public void scrollToElemnt(WebElement el) {
        JavascriptExecutor js = (JavascriptExecutor) driver;
        js.executeScript("arguments[0].scrollIntoView();", el);
        sleep(500);
	}
	
	public void scrollByPixel(int pixel) {
        JavascriptExecutor js = (JavascriptExecutor) driver;
        js.executeScript("window.scrollBy(0," + pixel + ")");
	}
	
}
